src/lib/network_pool/transaction_pool.ml

```diff
@@ -133,18 +178,30 @@ module Diff_versioned = struct
           "This transaction has expired"
       | Overloaded ->
           "The diff containing this transaction was too large"
+      | After_slot_tx_end ->
+          "This transaction was submitted after the slot defined to stop \
+           accepting transactions"
   end
 
   module Rejected = struct
     [%%versioned
     module Stable = struct
       [@@@no_toplevel_latest_type]
 
+      module V2 = struct
```

Comment:
Note for future: discuss on how this affects berkeley versioning types (whould we bump some types' versions there too?)

Reply:
This is the first time I'm updating versioned types, so I'm unsure on how this would affect 
`berkeley` types.
Regardless, this PR has merge conflicts with `berkeley` and `develop`, so we need to merge this feature into there as well. We can then upgrade these types there. Does this solve the issue?

Reply:
Yes, for sure. This is the way to proceed. My main point was that if we add a new numbered type here (`V2`) which existed in different form in `berkeley`, we probably should make sure to use `V3` in `berkeley`. I.e. if types between `compatible` and `berkeley` differ, `berkeley` should be +1 from `compatible`. That said, if you will find an existing `V3` in `berkeley` for a type that is `V2` in `compatible`, it's safe to change this type right away without introducing a new number.

Reply:
`berkeley` and `develop` are expected to have the same types, so once you remake this change against `berkeley`, there should be no problem with preparing a PR against `develop`.